src/elements/e-video.vue

```diff
@@ -1,94 +1,82 @@
 <template>
-  <div :class="b()">
-    <div :class="responsive ? b('responsive') : null">
-      <iframe
-        :class="responsive ? b('iframe', { responsive }) : b('iframe')"
-        :src="mapSrc"
-        :allowfullscreen="allowfullscreen"
-        :frameborder="frameborder.toString()"
-        :width="width.toString()"
-        :height="height.toString()"
-      ></iframe>
-    </div>
+  <div :class="b({ responsive: hasFixedSize })">
+    <iframe
+      :class="b('iframe', { responsive: hasFixedSize })"
+      :src="mapSrc"
+      :allowfullscreen="allowFullscreen"
+      :width="width"
+      :height="height"
+    ></iframe>
   </div>
 </template>
 
 <script lang="ts">
-
-/**
- * CONFIG
- *  baseUrl:
- *    - vimeo: "https://player.vimeo.com/video/
- *    - youtube: "https://www.youtube.com/embed/"
- *  controls:
- *    - youtube disables in src with ?controls=0
- *    - vimeo not able to disable controls
- *  size:
- *    - width & height can either be responsive or hard coded
- *    - youtube standard -> width="560", height="315"
- */
   import { PropType, defineComponent } from 'vue';
 
-  const videoUrlSources = {
-    youtube: 'https://www.youtube.com',
-    vimeo: 'https://vimeo.com/',
+  export enum VideoSource {
+    Vimeo = 'vimeo',
+    Youtube = 'youtube'
+  }
+
+  /**
+   * Specifies the allowed video base url.
+   */
+  const VideoUrlSource: { [key: string]: string | string[] } = {
+    [VideoSource.Youtube]: ['https://www.youtube.com', 'https://youtu.be'],
+    [VideoSource.Vimeo]: 'https://vimeo.com/',
```

Comment:
suggestion: I would have always used an array. This way you would not have to handle two possible cases.